File: pl-domain/doctor-service/src/main/java/com/polylab/doctors/domain/service/DoctorServiceImpl.java
package com.polylab.doctors.domain.service;

import com.jxareas.jxcore.domain.service.PersistentService;
import com.polylab.doctors.persistence.model.Doctor;
import com.polylab.doctors.persistence.repository.DoctorRepository;
import org.springframework.stereotype.Service;

@Service
public class DoctorServiceImpl extends PersistentService<Doctor, Integer> implements DoctorService {
    public DoctorServiceImpl(DoctorRepository repository) {
        super(repository);
    }
}

File: pl-domain/doctor-service/src/main/java/com/polylab/doctors/domain/service/DoctorServiceImpl.kt
package com.polylab.doctors.domain.service

import com.jxareas.efficax.domain.service.PersistentService
import com.polylab.doctors.persistence.model.Doctor
import com.polylab.doctors.persistence.repository.DoctorRepository
import org.springframework.stereotype.Service

@Service
class DoctorServiceImpl(repository: DoctorRepository) :
    PersistentService<Doctor, Int>(repository), DoctorService
